### PR TITLE
Connect with context

### PR DESCRIPTION
**Pull request by ewoutp**
Have Connect function with context argument

### CODE REVIEW

File: broker.go [hunk 1]
```diff
@@ -237,13 +237,18 @@ func Dial(ctx context.Context, nodeAddresses []string, conf BrokerConf) (*Broker
...
+				return nil, ctx.Err()
+			}
 		}
 
-		err := broker.refreshMetadata(ctx, proto.MetadataV0)
-
-		if err == nil {
+		if err := broker.refreshMetadata(ctx, proto.MetadataV0); err == nil {
 			return broker, nil
+		} else if err := ctx.Err(); err != nil {
```
Comment (neunhoef) [inline broker.go:250]: This seems to swallow the error from line 248, is this intended?

**Review by neunhoef — COMMENTED** [review]
Unfinished, but I had one question.

File: broker.go (hunk 1 above)
Comment (ewoutp) [reply broker.go:250]: Yes. It means: If there is an error and the context is canceled, ignore the error and return the cancelation

Comment (neunhoef) [reply broker.go:250]: OK.

**Pull request merged**